airbyte-commons-worker/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -515,8 +515,13 @@ public KubePodProcess(final boolean isOrchestrator,
       podBuilder = podBuilder.withServiceAccount("airbyte-admin").withAutomountServiceAccountToken(true);
     }
 
+    List<LocalObjectReference> pullSecrets = imagePullSecrets
+        .stream()
+        .map(imagePullSecret -> new LocalObjectReference(imagePullSecret))
+        .collect(Collectors.toList());
+
     final Pod pod = podBuilder.withTolerations(buildPodTolerations(tolerations))
-        .withImagePullSecrets(new LocalObjectReference(imagePullSecret)) // An empty string turns this into a no-op setting.
+        .withImagePullSecrets(pullSecrets) // An empty string turns this into a no-op setting.
```

Comment:
nit: comment should probably now say "An empty list turns this into a no-op setting."